### PR TITLE
Replace HttpStatusException by HttpException and update test suite to not validate the redirect

### PR DESCRIPTION
**Pull request by Sgitario**
- Replace HttpStatusException by HttpException: due to Quarkus is not using legacy API anylonger https://github.com/quarkusio/quarkus/commit/f33cc7472057914ec3fdbaa8dc7edc2b78b770d2
- Redirect support has been removed: see https://github.com/quarkusio/quarkus/commit/e94ac2b8a2c973b81a23498b288fba4f6df7e76a

### CODE REVIEW

**Review by pjgg — APPROVED** [review]

File: 020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java [hunk 1]
```diff
@@ -10,13 +10,15 @@
...
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.quarkus.qe.http.non_application.endpoint.HelloResource;
 import io.quarkus.test.QuarkusProdModeTest;
 
+@Disabled("2.x is no longer backward compatible and hence not redirecting endpoints")
```
Comment (Sgitario) [inline 020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java:21]: @pjgg if this test is not longer valid, should we delete it? wdyt?

**Comment by Sgitario** [comment]
@pjgg there were more required changes around the redirect of endpoints has been deleted in Quarkus 2.x, would you mind to take another look? Moreover, I asked you what do you think about the backward test in the module 020.

File: 020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java (hunk 1 above)
Comment (pjgg) [reply 020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java:21]: Yes, I think that should be removed

**Review by pjgg — CHANGES_REQUESTED** [review]

File: 020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java (hunk 1 above)
Comment (Sgitario) [reply 020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java]: Deleted

**Review by pjgg — APPROVED** [review]

**Pull request merged**